web/compose-compiler-integration/build.gradle.kts

```diff
@@ -0,0 +1,134 @@
+fun cloneTemplate(templateName: String, contentMain: String, contentLib: String): File {
+    val tempDir = file("${project.buildDir.absolutePath}/temp/cloned-$templateName")
+    tempDir.deleteRecursively()
+    tempDir.mkdirs()
+    file("${projectDir.absolutePath}/main-template").copyRecursively(tempDir)
+    // tempDir.deleteOnExit()
+    File("$tempDir/src/commonMain/kotlin/Main.kt").printWriter().use { out ->
+        out.println(contentMain)
+    }
+    File("$tempDir/lib/src/commonMain/kotlin/Lib.kt").printWriter().use { out ->
+        out.println(contentLib)
+    }
+    return tempDir
+}
+
+fun build(
+    caseName: String,
+    directory: File,
+    failureExpected: Boolean = false,
+    vararg buildCmd: String = arrayOf("build", "jsNodeRun")
+) {
+    val isWin = System.getProperty("os.name").startsWith("Win")
+    val procBuilder = if (isWin) {
+        ProcessBuilder("gradlew.bat", *buildCmd)
+    } else {
+        ProcessBuilder("bash", "./gradlew", *buildCmd)
+    }
+    val proc = procBuilder
+        .directory(directory)
+        .redirectOutput(ProcessBuilder.Redirect.PIPE)
+        .redirectError(ProcessBuilder.Redirect.PIPE)
+        .start()
+
+    proc.waitFor(5, TimeUnit.MINUTES)
+
+    "(COMPOSE_INTEGRATION_VERSION=\\[.*\\])".toRegex().find(
+        proc.inputStream.bufferedReader().readText()
+    )?.also {
+        println(it.groupValues[1])
+    }
+
+    println(proc.errorStream.bufferedReader().readText())
```

Comment:
should we print this all the time or only when exit code is not null?

Reply:
in case when exitCode == 0, it will simply print an empty string.
So yes, it's not necessary to always print, although it doesn't hurt. This printed message is only for our convenience when we read the output.